syntax = "proto3";

package projects;

service ProjectsService {
  rpc GetProjects(GetProjectsRequest) returns (GetProjectsResponse);
  rpc GetProjectById(GetProjectByIdRequest) returns (GetProjectByIdResponse);
  rpc GetProjectTree(GetProjectTreeRequest) returns (GetProjectTreeResponse);
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

message Project {
  int32 id = 1;
  string name = 2;
  string type = 3;
  optional string file_type = 4;
  optional int32 parent_id = 5;
  optional string url = 6;
  repeated Project children = 7;
  string created_at = 8;
  string updated_at = 9;
}

message GetProjectsRequest {
  optional int32 parent_id = 1;
  optional int32 depth = 2;
}

message GetProjectsResponse {
  repeated Project projects = 1;
}

message GetProjectByIdRequest {
  int32 id = 1;
  optional int32 depth = 2;
}

message GetProjectByIdResponse {
  optional Project project = 1;
}

message GetProjectTreeRequest {
  optional int32 root_id = 1;
}

message GetProjectTreeResponse {
  repeated Project projects = 1;
}

message HealthCheckRequest {}

message HealthCheckResponse {
  string status = 1;
  string timestamp = 2;
  bool database = 3;
  bool redpanda = 4;
  bool modules_service = 5;
}

